### PR TITLE
allow unregistering a factory

### PR DESCRIPTION
**Pull request by barakman**

### CODE REVIEW

**Review by lbeder — CHANGES_REQUESTED** [review]

File: solidity/contracts/converter/ConverterFactory.sol [hunk 1]
```diff
@@ -56,6 +56,42 @@ contract ConverterFactory is IConverterFactory, Owned {
...
+    /**
+     * @dev unregisters a specific typed converter factory
+     * can only be called by the owner
+     *
+     * @param _factory typed converter factory
+     */
+    function unregisterTypedConverterFactory(ITypedConverterFactory _factory) public ownerOnly {
+        uint16 converterType = _factory.converterType();
+        require(converterFactories[converterType] == _factory, "ERR_NOT_REGISTERED");
+        converterFactories[converterType] = ITypedConverterFactory(0);
```
[the remark this answers is not in the document]
Comment (barakman) [reply solidity/contracts/converter/ConverterFactory.sol]: fixed

File: solidity/contracts/helpers/TestTypedConverterCustomFactory.sol [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+import "../converter/interfaces/ITypedConverterCustomFactory.sol";
```
[the remark this answers is not in the document]
Comment (barakman) [reply solidity/contracts/helpers/TestTypedConverterCustomFactory.sol:3]: fixed

File: solidity/contracts/converter/ConverterFactory.sol [hunk 3]
```diff
@@ -27,7 +27,7 @@ contract ConverterFactory is IConverterFactory, Owned {
     mapping(uint16 => ITypedConverterCustomFactory) public override customFactories;
 
     /**
-     * @dev initializes the factory with a specific typed converter factory
+     * @dev registers a specific typed converter factory
```
[the remark this answers is not in the document]
Comment (barakman) [reply solidity/contracts/converter/ConverterFactory.sol:31]: fixed

File: solidity/contracts/converter/ConverterFactory.sol [hunk 4]
```diff
@@ -56,6 +56,42 @@ contract ConverterFactory is IConverterFactory, Owned {
         customFactories[_factory.converterType()] = _factory;
     }
 
+    /**
+     * @dev unregisters a specific typed converter factory
+     * can only be called by the owner
+     *
+     * @param _factory typed converter factory
+     */
+    function unregisterTypedConverterFactory(ITypedConverterFactory _factory) public ownerOnly {
```
[the remark this answers is not in the document]
Comment (barakman) [reply solidity/contracts/converter/ConverterFactory.sol]: fixed

File: solidity/contracts/converter/ConverterFactory.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (lbeder) [reply solidity/contracts/converter/ConverterFactory.sol:31]: 💪

File: solidity/contracts/converter/ConverterFactory.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (lbeder) [reply solidity/contracts/converter/ConverterFactory.sol]: 💪

File: solidity/contracts/converter/ConverterFactory.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (lbeder) [reply solidity/contracts/converter/ConverterFactory.sol:31]: Thanks for going over it. There are only two changes left:
1. State variables should be only `private` (and explicit getters need to be added if/where required).
2. There's no need for padding argument descriptions in the NATSPEC.

[the remark this answers is not in the document]
Comment (barakman) [reply solidity/contracts/converter/ConverterFactory.sol:64]: done

**Review by lbeder — APPROVED** [review]
Nice job! 👍

**Pull request merged**